### PR TITLE
Add support for base64-encoded hash values to 'get_file_hash'

### PR DESCRIPTION
**Pull request by SkypLabs**
**IMPORTANT: Please do not create a Pull Request adding a new feature without discussing it first.**

The place to discuss new features is the forum: <https://zola.discourse.group/>
If you want to add a new feature, please open a thread there first in the feature requests section.

Sanity check:

* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/getzola/zola/pulls) for the same update/change?

## Code changes
(Delete or ignore this section for documentation changes)

* [x] Are you doing the PR on the `next` branch?

If the change is a new feature or adding to/changing an existing one:

* [x] Have you created/updated the relevant documentation page(s)?

---

This PR followings up with the discussion in https://github.com/getzola/zola/issues/519.

### CODE REVIEW

File: components/templates/src/global_fns/mod.rs [hunk 1]
```diff
@@ -95,18 +96,36 @@ fn compute_file_sha256(mut file: fs::File) -> result::Result<String, io::Error>
     Ok(format!("{:x}", hasher.finalize()))
 }
 
+fn compute_file_sha256_base64(mut file: fs::File) -> result::Result<String, io::Error> {
```
Comment (Keats) [inline components/templates/src/global_fns/mod.rs:99]: I think it will be cleaner to pass the base64 arg to those functions than to duplicate them all.

Comment (SkypLabs) [reply components/templates/src/global_fns/mod.rs:99]: I thought about it but there were already 3 different functions for the 3 possible values of `sha_type` so I opted for keeping it consistent.

Wouldn't it be even better to pass `sha_type` and `base64` to one unique function handling all the possible combinations?

Comment (Keats) [reply components/templates/src/global_fns/mod.rs:99]: > Wouldn't it be even better to pass sha_type and base64 to one unique function handling all the possible combinations?

Even better!

**Comment by SkypLabs** [comment]
Documentation updated. I also fixed the SRI issue in `test_site`.

**Comment by Keats** [comment]
Is it still a WIP somehow or ok to merge?

**Comment by SkypLabs** [comment]
Hey @Keats.

I didn't find the time to implement your suggestion of merging the different hashing functions into a single one (see https://github.com/getzola/zola/pull/1339#discussion_r570968484). However, my PR can be merged in its current state so it's up to you.

**Comment by Keats** [comment]
I'll fix it post-merge, thanks!

**Comment by Keats** [comment]
https://github.com/getzola/zola/pull/1293/commits/3262f69bb06b619d333aacae7cb0e5abe8d497e5

**Comment by SkypLabs** [comment]
Awesome! Thanks

**Pull request merged**